arkouda/io.py

```diff
@@ -1912,3 +1913,43 @@ def read_tagged_data(
         raise RuntimeError("CSV does not support tagging data with file name associated.")
     else:
         raise RuntimeError(f"Invalid File Type detected, {ftype}")
+
+
+def receive_array(hostname : str, port):
+    """
+    Receive a pdarray sent by `pdarray.send_array()`.
+
+    Parameters
+    ----------
+    hostname : str
+        The hostname of the pdarray that sent the array
+    port : int_scalars
+        The port to send the array over. This needs to be an
+        open port (i.e., not one that the Arkouda server is
+        running on). This will open up `numLocales` ports,
+        each of which in succession, so will use ports of the
+        range {port..(port+numLocales)} (e.g., running an
+        Arkouda server of 4 nodes, port 1234 is passed as
+        `port`, Arkouda will use ports 1234, 1235, 1236,
+        and 1237 to send the array data).
```

Comment:
would these ports need to be the same or distinct from those in `send_array`? I'm guessing distinct (or at least not necessarily the same) because we could have a different number of locales between the receiving and sending servers right?

Reply:
It will be the number of ports from the server with more locales, so, if we had a server sending with 4 locales and only receiving on 2, both of the servers will open up the 4 ports (since those are the ports that were sent on).

Reply:
cool thanks for clarifying!